Fix action column width on D&D tables

jpople opened the pull request:
Closes HA-179

### Description Of Changes

Fixes a bug where D&D action buttons were sometimes cut off at certain screen sizes and after resizing other columns.

https://github.com/user-attachments/assets/1206a8f2-170f-4279-bb03-7a1302044c40

### Code Changes

* Changes all "Actions" columns in D&D tables to have width `auto`
* Also removes stopPropagation calls to move to updated `disableRowClick`pattern

### Steps to Confirm

1. Populate the D&D tables with some items
2. On both detection tables and discovery tables, make sure action buttons are visible and don't get cut off when resizing

### Pre-Merge Checklist

* [x] Issue requirements met
* [ ] All CI pipelines succeeded
* [ ] `CHANGELOG.md` updated
  * [ ] Add a https://github.com/ethyca/fides/labels/db-migration label to the entry if your change includes a DB migration
  * [ ] Add a https://github.com/ethyca/fides/labels/high-risk label to the entry if your change includes a high-risk change (i.e. potential for performance impact or unexpected regression) that should be flagged
* Followup issues:
  * [ ] Followup issues created (include link)
  * [x] No followup issues
* Database migrations:
  * [ ] Ensure that your downrev is up to date with the latest revision on `main`
  * [ ] Ensure that your `downgrade()` migration is correct and works
    * [ ] If a downgrade migration is not possible for this change, please call this out in the PR description!
  * [x] No migrations
* Documentation:
  * [ ] Documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] Documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
  * [ ] If there are any new client scopes created as part of the pull request, remember to update public-facing documentation that references our scope registry
  * [x] No documentation updates required

lucanovera reviewed (commented):
Thanks for fixing the stop propagation too, disableRowClick is the way to go. I've tested in Chrome and Firefox and it's not cutting off anymore. 

The variable auto width should be the more correct option, but for some reason it's making the column fixed and too wide for just the two buttons. Can we make a last effort to try to reduce that space before merging the fix? I've played around width using size: 250 instead of 180.

jpople commented on clients/admin-ui/src/features/data-discovery-and-detection/hooks/useDiscoveryResultColumns.tsx:
```diff
@@ -17,6 +17,9 @@ import DiscoveryItemActionsCell from "../tables/cells/DiscoveryItemActionsCell";
 import ResultStatusCell from "../tables/cells/ResultStatusCell";
 
+const NAME_COLUMN_SIZE = 300;
+const ACTION_COLUMN_SIZE = 220;
```
This one is slightly wider since it potentially needs to accommodate a kebab menu also.

lucanovera replied:
```suggestion
const ACTION_COLUMN_SIZE = 235;
```
Good catch! I did found that if I intentional shink the column down to it's minimum width it will still generate a bit of scrolling. Bringing it up a to 235 removes that scroll.

<img width="377" alt="Captura de pantalla 2025-02-26 a la(s) 2 57 04 p  m" src="https://github.com/user-attachments/assets/a434ae6d-ec5f-43af-b8ee-811e924c86f0" />

jpople commented:
@lucanovera The column width behavior is pretty confusing and I think we've been using `width: auto` wrong in a few places-- putting `auto` on any column essentially forces all the non-auto columns to be their minimum sizes and allocates the rest of the space to the `auto` column(s).  

I've removed `auto` from the actions column in the discovery tables and added static widths, which has largely fixed the excess space issue; it doesn't seem feasible to guarantee zero excess space, but at least it won't be taking up the entire table now.

lucanovera commented:
> @lucanovera The column width behavior is pretty confusing and I think we've been using `width: auto` wrong in a few places-- putting `auto` on any column essentially forces all the non-auto columns to be their minimum sizes and allocates the rest of the space to the `auto` column(s).
> 
> I've removed `auto` from the actions column in the discovery tables and added static widths, which has largely fixed the excess space issue; it doesn't seem feasible to guarantee zero excess space, but at least it won't be taking up the entire table now.

Thanks @jpople . I'm testing and it's looking great. No cutoffs or excess spaces.

lucanovera reviewed (approved):
Looking great! Tested in Chrome and Firefox, switching resolutions and column widths. Code looks good. Approved!
